From e79d1afa9d2c0ec8990eecc7623ac1f540c6e4ac Mon Sep 17 00:00:00 2001
From: Janos Kovacs <janos.kovacs@intel.com>
Date: Sun, 6 Sep 2015 08:40:11 +0300
Subject: [PATCH 2/3] rpm5: Fix rpmcliFini that subsequent cli creation work
 properly

In application framework we need to make multiple cli operations,
eg. an 'rpm -qpl' equivalent to check content of the package and
if everything is found correct, install the package with a 'rpm -i'
equivalent.

Due to a bug the second cli operation always failed as the first
one was not fully wind-down and the second operation was launched
with an empty configuration (ie. no popt macros were defined etc.)

This patch fixes the bug and lets the cli operations wind-down
properly, leaving a clean environment for the subsecquent cli's.

Upstream-Status: Pending

Signed-off-by: Janos Kovacs <janos.kovacs@intel.com>
---
 lib/poptALL.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/lib/poptALL.c b/lib/poptALL.c
index aa129a5..a0d9430 100644
--- a/lib/poptALL.c
+++ b/lib/poptALL.c
@@ -591,6 +591,7 @@ rpmcliFini(poptContext optCon)
 /*@i@*/	rpmFreeMacros(rpmCLIMacroContext);
 
     rpmFreeRpmrc();	/* XXX mireFreeAll(platpat) before rpmioFreePool. */
+    rpmcliInitialized = -1;
 
     rpmFreeFilesystems();
 /*@i@*/	rpmcliTargets = _free(rpmcliTargets);
-- 
2.1.0

